src/main/webapp/app/overview/course-archive/course-archive.component.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Course } from 'app/entities/course.model';
+import { CourseManagementService } from '../../course/manage/course-management.service';
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { AlertService } from 'app/core/util/alert.service';
+import { onError } from 'app/shared/util/global.utils';
+import { Subscription } from 'rxjs';
+import { faAngleDown, faAngleUp, faArrowDownAZ, faArrowUpAZ, faQuestionCircle } from '@fortawesome/free-solid-svg-icons';
+import { sortCourses } from 'app/shared/util/course.util';
+import { SizeProp } from '@fortawesome/fontawesome-svg-core';
+import { ArtemisSharedModule } from 'app/shared/shared.module';
+import { ArtemisSharedComponentModule } from 'app/shared/components/shared-component.module';
+import { CourseCardHeaderComponent } from '../course-card-header/course-card-header.component';
+
+@Component({
+    selector: 'jhi-course-archive',
+    templateUrl: './course-archive.component.html',
+    styleUrls: ['./course-archive.component.scss'],
+    standalone: true,
+    imports: [ArtemisSharedModule, ArtemisSharedComponentModule, CourseCardHeaderComponent],
```

Comment:
Do you really need all those dependencies here? Maybe it's possible to just remove those imports, and only import smaller modules which you actually need

Reply:
We can import only the required components but the problem is these components need to be also standalone. SearchComponent and its pipe does not standalone and therefore I don't want to change them into standalone in this pr, since it will also affect shared modules.